tests/unit/package_managers/test_pip.py

```diff
@@ -105,23 +112,25 @@ def test_get_pip_metadata(
     expect_name = toml_name or py_name or cfg_name
     expect_version = toml_version or py_version or cfg_version
 
-    if expect_name and expect_version:
+    if expect_name:
         name, version = pip._get_pip_metadata(PKG_DIR)
 
         assert name == expect_name
         assert version == expect_version
     else:
-        with pytest.raises(PackageRejected) as exc_info:
-            pip._get_pip_metadata(PKG_DIR)
-
-        if expect_name:
-            missing = "version"
-        elif expect_version:
-            missing = "name"
+        if repo_exists:
+            mock_get_repo_id.return_value = MOCK_REPO_ID
+            name, _ = pip._get_pip_metadata(PKG_DIR_SUBPATH)
+            assert name == "bar-subpath"
+            assert "Processing metadata from git repository" in caplog.text
         else:
-            missing = "name, version"
-
-        assert str(exc_info.value) == f"Could not resolve package metadata: {missing}"
+            mock_get_repo_id.side_effect = PackageRejected(
+                reason="Could not take name from the repository origin url", solution=None
+            )
```

Comment:
get_repo_id raises UnsupportedFeature, not PackageRejected

Because of that, the code that that raises PackageRejected is not covered https://coveralls.io/builds/61558868/source?filename=cachi2%2Fcore%2Fpackage_managers%2Fpip.py#L213